tools/custom-checks/bump-openapi-spec-dep-version.ts

```diff
@@ -0,0 +1,161 @@
+import fs from "fs-extra";
+import path from "path";
+import { fileURLToPath } from "url";
+import { globby, Options as GlobbyOptions } from "globby";
+import { RuntimeError } from "run-time-error";
+import { isStdLibRecord } from "./is-std-lib-record";
+import { exec } from "child_process";
+import util from "util";
+
+interface paths {
+  post: {
+    requestBody: unknown;
+  };
+}
+interface packageJson {
+  info: {
+    version: string;
+  };
+  paths: {
+    [key: string]: paths;
+  };
+}
+
+export interface IBumpOpenAPISpecDependencyVersionRequest {
+  readonly argv: string[];
+  readonly env: NodeJS.ProcessEnv;
+  readonly version?: string;
+}
+
+/**
+ * Verifies that each openapi.json is up to date with the latest version
+ * of the main/upstream branch.
+ * Lists all the files that are being found.
+ * And also logs all the total count of how many references (since each
+ * file can have any number of references in it).
+ */
+
+export async function bumpOpenAPISpecDependencyVersion(
+  req: IBumpOpenAPISpecDependencyVersionRequest,
+): Promise<[boolean, string[]]> {
+  const TAG = "[tools/bumpOpenAPISpecDependencyVersion.ts]";
+  /** change tag*/ const __filename = fileURLToPath(import.meta.url);
+  const __dirname = path.dirname(__filename);
+  const SCRIPT_DIR = __dirname;
+  const PROJECT_DIR = path.join(SCRIPT_DIR, "../../");
+  console.log(`${TAG} SCRIPT_DIR=${SCRIPT_DIR}`);
+  console.log(`${TAG} PROJECT_DIR=${PROJECT_DIR}`);
+
+  if (!req) {
+    throw new RuntimeError(`req parameter cannot be falsy.`);
+  }
+  if (!req.argv) {
+    throw new RuntimeError(`req.argv cannot be falsy.`);
+  }
+  if (!req.env) {
+    throw new RuntimeError(`req.env cannot be falsy.`);
+  }
+
+  const globbyOpts: GlobbyOptions = {
+    cwd: PROJECT_DIR,
+    ignore: ["**/node_modules"],
+  };
+
+  const DEFAULT_GLOB = "**/src/main/json/openapi.json";
+  /** cannot be empty */
+  const pkgJsonPaths = await globby(DEFAULT_GLOB, globbyOpts);
+
+  const execPromise = util.promisify(exec);
+  const repoUrl = "https://github.com/hyperledger/cacti";
+  const branchName = "main";
+  const latestVersion = execPromise(
+    `git describe --tags --abbrev=0 ${branchName}`,
+  ).then(({ stdout }) => {
+    const latestTag = stdout.trim().substring(1);
+    console.log(
+      `${TAG} Hyperledger Cacti current version of ${branchName} branch  ${repoUrl} is version ${latestTag}`,
+    );
+    return latestTag;
+  });
+
+  const currentLatestVersion = await latestVersion;
```

Comment:
@aldousalvarez It's probably the best to extract the entire logic of obtaining the latest git release tag into a function of it's own so that it could potentially be re-used elsewhere in the tooling code as well.